ui/tutorial-renderer/src/main/java/org/cru/godtools/tutorial/activity/TutorialActivity.kt

```diff
@@ -0,0 +1,156 @@
+package org.cru.godtools.tutorial.activity
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.viewpager.widget.ViewPager
+import me.relex.circleindicator.CircleIndicator
+import org.cru.godtools.base.Settings
+import org.cru.godtools.tutorial.PageSet
+import org.cru.godtools.tutorial.R
+import org.cru.godtools.tutorial.adapter.TutorialPagerAdapter
+import org.cru.godtools.tutorial.util.TutorialCallbacks
+
+private const val ARG_PAGE_SET = "pageSet"
+
+fun Activity.startTutorialActivity(pageSet: PageSet = PageSet.DEFAULT) {
+    Intent(this, TutorialActivity::class.java)
+        .putExtra(ARG_PAGE_SET, pageSet)
+        .also { startActivity(it) }
+}
+
+class TutorialActivity : AppCompatActivity(), TutorialCallbacks {
+    private val pageSet get() = intent?.getSerializableExtra(ARG_PAGE_SET) as? PageSet ?: PageSet.DEFAULT
+
+    // region Lifecycle
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.tutorial_activity)
+        setUpToolbar()
+    }
+
+    override fun onContentChanged() {
+        super.onContentChanged()
+        setupViewPager()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        pageSet.feature?.let { Settings.getInstance(this).setFeatureDiscovered(it) }
+    }
+
+    private var tutorialMenu: Menu? = null
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        if (pageSet == PageSet.ONBOARDING) {
+            menuInflater.inflate(R.menu.tutorial_menu, menu)
```

Comment:
instead of hardcoding a specific menu here, you should add a property to the `PageSet` that defines the menu res to inflate. And then you only inflate a menu if one is defined for the pageSet

Reply:
it simplifies some of your menu customization, unfortantly you will still need to manually hide the menu for the first page of the onboarding tutorial